Food_Tracker_App/app/src/main/java/com/example/foodtracker/ui/mealPlan/MealPlanMainScreen.java

```diff
@@ -21,13 +58,288 @@ public class MealPlanMainScreen extends AppCompatActivity {
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.meal_plan_main);
+        initializeSort();
+        //initializeData();
+
+        if (savedInstanceState == null){
+            createRecyclerView();
+            createNavBar();
+            createTopBar();
+        }
+
+        if (getIntent().getExtras() != null) {
+            Intent intent = getIntent();
+            if (intent.getSerializableExtra(MEAL_PLAN_AFTER_INGREDIENT_ADD) != null) {
+                MealPlanDay received_meal_plan = (MealPlanDay) intent.getSerializableExtra(MEAL_PLAN_AFTER_INGREDIENT_ADD);
+                addIngredient(received_meal_plan);
+            }
+            if (intent.getSerializableExtra(MEAL_PLAN_AFTER_RECIPE_ADD) != null) {
+                MealPlanDay received_meal_plan = (MealPlanDay) intent.getSerializableExtra(MEAL_PLAN_AFTER_RECIPE_ADD);
+                addRecipe(received_meal_plan);
+            }
+            /**
+             * edit recipe
+             */
+            if (intent.getSerializableExtra("meal_plan_after_recipe_edit") != null) {
+                MealPlanDay received_meal_plan = (MealPlanDay) intent.getSerializableExtra("meal_plan_after_recipe_edit");
+                editRecipe(received_meal_plan);
+            }
+        }
+
+    }
+
+    /**
+     * Adds some initial data to the list
+     */
+    private void initializeData() {
+        mealPlanDaysCollection.getAll(list -> {
+            mealPlanDayArrayList.addAll(list);
+            adapter.notifyItemRangeInserted(0, mealPlanDayArrayList.size());
+        });
+    }
```

Comment:
I believe you only want to initialize sort, since sort will already populate this list

Reply:
Nevermind it is commented out, just delete this then